applications/fishing-map/src/features/resources/resources.selectors.ts

```diff
@@ -0,0 +1,95 @@
+import { createSelector } from 'reselect'
+import {
+  Dataset,
+  DatasetTypes,
+  DataviewDatasetConfig,
+  EndpointId,
+} from '@globalfishingwatch/api-types'
+import { resolveEndpoint } from '@globalfishingwatch/dataviews-client'
+import { Generators } from '@globalfishingwatch/layer-composer'
+import { ResourceQuery } from 'features/resources/resources.slice'
+import { selectDataviewInstancesResolved } from 'features/workspace/workspace.selectors'
+import { UrlDataviewInstance } from 'types'
+import { isGuestUser } from 'features/user/user.selectors'
+import { selectDebugOptions } from 'features/debug/debug.slice'
+import { ThinningLevels, THINNING_LEVELS } from 'data/config'
+
+export const resolveDataviewDatasetResource = (
+  dataview: UrlDataviewInstance,
+  { type, id }: { type?: DatasetTypes; id?: string }
```

Comment:
I know that code is not new, but it's hard to understand just looking at the function signature that it's either `type` OR `id`

```suggestion
 typeOrId: DatasetTypes | string```
 
 Then, I guess we should look at https://medium.com/ovrsea/checking-the-type-of-an-object-in-typescript-the-type-guards-24d98d9119b0